test/config-luatest/config_test.lua

```diff
@@ -1393,3 +1393,180 @@ g.test_non_dynamic_alert = function(g)
         end,
     })
 end
+
+-- Check that config:info('v2') works properly.
+g.test_info_new_version = function(g)
+    local dir = treegen.prepare_directory(g, {}, {})
+
+    local config = [[
+        credentials:
+          users:
+            guest:
+              roles:
+              - super
+
+        iproto:
+          listen:
+            - uri: unix/:./{{ instance_name }}.iproto
+
+        groups:
+          group-001:
+            replicasets:
+              replicaset-001:
+                instances:
+                  instance-001: {}
+    ]]
+    treegen.write_script(dir, 'config.yaml', config)
+
+    -- New source that sets meta.
+    local script = string.dump(function()
+        local yaml = require('yaml')
+        local file = require('internal.config.utils.file')
+
+        local methods = {}
+        local mt = {
+            __index = methods,
+        }
+
+        function methods.sync(self, config)
+            local data = file.universal_read('myconf.yaml', 'my config file')
+            local ok, res = pcall(yaml.decode, data)
+            if not ok then
+                error('Unable to parse my config file')
+            end
+            self._values = res
+            config:_meta('myconf', 'revision', self._revision)
```

Comment:
Nit: It is not important for testing, but if we take it as example of a configuration source with metainformation, I would use the same `:_meta()` key as the source name.

Reply:
True, fixed.

Reply:
Sorry for nitpicking, but the source still named as `myfile` (in the `name` field), while placed into `mysource.lua` and reported as `mysource` in the metainformation.

Reply:
Sorry, fixed.